docs/content/en/docs/tutorials/ci_cd.md

```diff
@@ -0,0 +1,286 @@
+
+
+## Step 6: Development & Testing locally
+
+Before we move on to setting up the CI pipeline, we should test it out locally. Use **skaffold run** if you are deploying it to the cluster, **skaffold build **to just build and push the artifacts, **skaffold dev **if you are developing (this will enable auto reload on changes) or debug using **skaffold debug**
+
+You can find docs about all these workflows here: [https://skaffold.dev/docs/workflows/](https://skaffold.dev/docs/workflows/)
+
+Also, note that you can also set up file synchronization to enable faster development workflows. You can read more about that here: [https://skaffold.dev/docs/pipeline-stages/filesync/](https://skaffold.dev/docs/pipeline-stages/filesync/)
+
+
+## Step 7: Getting the CI pipeline setup
+
+Now that we have skaffold ready and working locally, the CI pipeline requires little to no effort since it will just call skaffold to do all the builds and deployments.
+
+Here is a sample pipeline using Gitlab CI
+
+
+```
+stages:
+    - my-stage
+
+my-job:
+    image:
+        name: gcr.io/k8s-skaffold/skaffold:v1.15.0
+    tags:
+        - development
+    stage: my-stage
+    retry: 2
+    script:
+        - docker login $REG_DEV_URL -u $REG_DEV_UNAME -p $REG_DEV_PASSWORD
+        - echo "$GCP_DEV_SERVICE_KEY" > gcloud-service-key.json
+        - gcloud auth activate-service-account --key-file gcloud-service-key.json
+        - gcloud config set project $PROJECT_DEV_NAME
+        - gcloud config set compute/zone $PROJECT_DEV_REGION
+        - gcloud container clusters get-credentials $CLUSTER_DEV_NAME
+        - kubectl config get-contexts
+        - DOCKER_HOST="$DIND_DEV" NPM_REGISTRY="$NPM_REGISTRY_DEV" NPM_TOKEN="$NPM_TOKEN_DEV" skaffold run --kube-context $CLUSTER_DEV_CONTEXT -n default -l skaffold.dev/run-id=deploydep -p dev-svc --status-check
+    only:
+        refs:
+            - master
+```
+
+
+Before we look at the pipeline, you might want to look at Gitlab CI/CD environment variables here: [https://docs.gitlab.com/ee/ci/variables/](https://docs.gitlab.com/ee/ci/variables/)
+
+This is how we will configure the various environment variables to be passed in to the pipeline. Make sure that you mask/protect them using Gitlab so that they are not visible in the logs.
+
+Here, we are:
+
+
+
+*   Logging into the docker registry specifying the Registry URL, User ID and password (You can choose to authenticate using service tokens as well or other auth providers as well)
+*   Logging in to our gcp account using the service account key
+*   Specifying the project, zone, cluster to deploy our application in.
+*   We are using DIND (Docker In Docker) to facilitate our builds (more about that below)
+*   Then we just call the skaffold run in the pipeline passing flags we want to pass at runtime (in our case our cluster context, env vars, namespace and some labels)
+*   And we trigger this only on changes to master branch
```

Comment:
@briandealwis Thanks a lot for your suggestions.  Sure. Can change this. Yes, I use a private registry (Verdaccio) and that is what it authenticates with - it can be avoided, but I just put it here so that people would understand how to pass in args from gitlab to skaffold which then gets passed to the build args in Docker.